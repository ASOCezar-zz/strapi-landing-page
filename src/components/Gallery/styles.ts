import styled, { css, keyframes } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.font.spacings.huge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.font.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

const bordersAnimation = keyframes`
  0%{border-radius: 100%; transform: scale(0)};
  50%{border-radius: 50%; transform: scale(1.5)}
  100%{border-radius: 0; transform: scale(1)};
`;

const animateElement = ({ number, show }) =>
  show &&
  css`
    animation: ${bordersAnimation} 500ms ${number}s ease forwards;
  `;

export const GridElement = styled.div`
  ${({ number, show }: GridType) => css`
    overflow: hidden;
      ${animateElement({ number, show })}
    }
  `}
`;

const showItem = keyframes`
  0% {transform: scale(0)};
  50% {transform: scale(1.5)};
  100% {transform: scale(1)}
`;

type GridType = {
  number: number;
  show: boolean;
};

const animateGrid = ({ number, show }: GridType) =>
  show &&
  css`
    animation: ${showItem} 350ms ${number}s ease forwards;
  `;

export const Image = styled.img<GridType>`
  ${({ number, show }) => css`
    transform: scale(0);
    width: 100%;

    ${animateGrid({ number, show })}
  `}
`;
