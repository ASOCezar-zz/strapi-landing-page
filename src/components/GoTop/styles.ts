import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background-color: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 5rem;
    height: 5rem;

    bottom: 3rem;
    right: 2rem;

    z-index: 5;
    opacity: 0.6;

    border-radius: 50%;
  `}
`;
