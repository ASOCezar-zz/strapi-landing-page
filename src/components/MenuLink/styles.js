import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    font-size: ${theme.font.sizes.small};
    padding: ${theme.font.spacings.small};
    color: ${theme.colors.primaryColor};
    position: relative;

    &::after {
      content: '';
      position: absolute;
      bottom: 0.78rem;
      width: 0;
      height: 0.2rem;
      background: ${theme.colors.secondaryColor};
      left: 50%;
      transition: all 0.2s ease-in-out;
    }

    &:hover::after {
      width: 50%;
      left: 25%;
    }
  `}
`;
