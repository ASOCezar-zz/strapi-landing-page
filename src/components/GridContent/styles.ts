import styled, { css, keyframes } from 'styled-components';

import { Title } from '../Heading/styles';

export type ContainerType = {
  show: boolean;
};

const showTitle = keyframes`
  to {transform: translateY(0)};
`;

const presentContent = ({ show }: ContainerType) =>
  show &&
  css`
    ${Title} {
      animation: ${showTitle} 300ms 1s ease-in-out forwards;
    }
  `;

export const Container = styled.div<ContainerType>`
  ${({ show }) => css`
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
    overflow: hidden;
    transition: all 1s 1s ease-in-out;

    ${Title} {
      transform: translateY(-200%);
    }

    ${presentContent({ show })};
  `}
`;

const showText = keyframes`
  to {opacity: 1};
`;

const presentText = ({ show }: ContainerType) =>
  show &&
  css`
    animation: ${showText} 300ms 1500ms ease-in-out forwards;
  `;

export const Html = styled.div<ContainerType>`
  ${({ theme, show }) => css`
    margin-block: ${theme.font.spacings.huge};
    opacity: 0;

    ${presentText({ show })};
  `}
`;
