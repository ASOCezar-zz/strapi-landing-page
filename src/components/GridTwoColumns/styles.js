import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme, background }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.font.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    ${Title} {
      margin-bottom: ${theme.font.spacings.xlarge};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.font.spacings.large};
    }
  `}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css``}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
  `}
`;
