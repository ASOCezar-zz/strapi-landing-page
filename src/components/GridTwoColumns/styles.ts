import styled, { css, keyframes } from 'styled-components';
import { Title } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

const shownTitle = keyframes`
  to{transform: translateX(0)};
`;

const shownDescription = keyframes`
  to{opacity: 1};
`;

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.font.spacings.large};
    tr @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    ${Title} {
      margin-bottom: ${theme.font.spacings.xlarge};
      transform: translateX(-100vw);
      animation: ${shownTitle} 300ms linear forwards;
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      opacity: 0;

      animation: ${shownDescription} 250ms 500ms linear forwards;
    }

    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.font.spacings.large};
    }
  `}
`;

export const ImageContainer = styled.div``;

const hasShown = keyframes`
  to {transform: scale(1)};
`;

export const Image = styled.img`
  ${() => css`
    width: 100%;
    transform: scale(0);
    animation: ${hasShown} 300ms 500ms ease-in-out forwards;
  `}
`;
