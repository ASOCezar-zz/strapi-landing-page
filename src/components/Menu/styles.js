import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';
import { Title as Heading } from '../Heading/styles';

const menuVisible = () => css`
  visibility: visible;
  opacity: 1;
`;

export const Container = styled.header`
  ${({ theme, isMenuVisible }) => css`
    position: fixed;
    z-index: 5;
    left: 0;
    right: 0;
    top: 0;
    transition: all 0.3s ease-in-out;

    border-bottom: 0.1rem ${theme.colors.mediumGray};
    background: ${theme.colors.white};

    & ${SectionContainer} {
      padding-block: 0;
    }

    & ${Heading} {
      margin-block: 0;
    }

    @media ${theme.media.lteMedium} {
      height: 100vh;
      visibility: hidden;
      opacity: 0;

      ${isMenuVisible && menuVisible(theme)}

      & ${SectionContainer} {
        display: grid;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr;
        height: 100vh;
        align-items: center;
        margin: 0;
        overflow-y: auto;
        height: 100%;
        width: 100%;
        margin: 0;
        padding: 0;
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;

    @media ${theme.media.lteMedium} {
      display: block;
      text-align: center;
      padding: ${theme.font.spacings.huge} 0;
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    z-index: 6;
    position: fixed;
    top: 3rem;
    right: 2rem;
    width: 4rem;
    height: 4rem;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    border: none;
    display: none;

    > svg {
      width: 2.5rem;
      height: 2.5rem;
    }

    @media ${theme.media.lteMedium} {
      display: flex;
      align-items: center;
      justify-content: center;
      cursor: pointer;
    }
  `}
`;
