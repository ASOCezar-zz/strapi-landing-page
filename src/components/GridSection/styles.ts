import styled, { css, keyframes } from 'styled-components';
import { ContainerType } from '../GridContent/styles';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

const revealText = keyframes`
  from {transform: translateX(-101%)}
  to {transform: translateX(101%)}
`;

const fade = keyframes`
  to{opacity: 1};
`;

const revealTitle = ({ show }: ContainerType) =>
  show &&
  css`
    ::after {
      animation: ${revealText} 500ms ease-out forwards;
    }
    > ${HeadingContainer} {
      animation: ${fade} 320ms 600ms linear forwards;
    }
  `;

const revealDescription = ({ show }: ContainerType) =>
  show &&
  css`
    animation: ${fade} 320ms 600ms linear forwards;
  `;

export const Container = styled.div<ContainerType>`
  ${({ theme, show }) => css`
    position: relative;

    .reveal-container {
      position: relative;
      overflow: hidden;
      width: fit-content;
      ${revealTitle({ show })}
      ::after {
        content: '';
        position: absolute;
        background-color: white;
        inset: 0;
        display: block;
      }

      ${HeadingContainer} {
        opacity: 0;
      }
    }

    > ${TextComponent} {
      opacity: 0;
      ${revealDescription({ show })}
    }

    & ${TextComponent} {
      margin-bottom: ${theme.font.spacings.medium};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.font.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

const showItem = keyframes`
  from {transform: scale(0)}
  to {transform: scale(1)};
`;

type GridType = {
  number: number;
  show: boolean;
};

const animateGrid = ({ number, show }: GridType) =>
  show &&
  css`
    animation: ${showItem} 350ms ${number}s linear forwards;
  `;

export const GridElement = styled.div<GridType>`
  ${({ theme, number, show }) => css`
    transform: scale(0);
    ${animateGrid({ number, show })}
    & ${HeadingContainer} {
      position: relative;
      left: -7rem;

      @media ${theme.media.lteMedium} {
        left: 3.1rem;
      }
    }

    & ${HeadingContainer}:before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.font.sizes.huge};
      top: -2rem;
      left: -5rem;
      transform: rotate(12deg);
    }
  `}
`;
