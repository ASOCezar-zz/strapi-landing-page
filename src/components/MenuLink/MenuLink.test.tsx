import { screen } from '@testing-library/react';
import { MenuLink } from '.';
import { renderTheme } from '../../styles/renderTheme';

describe('<MenuLink />', () => {
  it('Should render a link', () => {
    renderTheme(
      <MenuLink link="https://google.com" newTab={false}>
        children
      </MenuLink>,
    );
    expect(screen.getByRole('link', { name: 'children' })).toBeInTheDocument;
  });

  it('Should render a link', () => {
    renderTheme(
      <MenuLink link="https://google.com" newTab={true}>
        children
      </MenuLink>,
    );
    expect(screen.getByRole('link', { name: 'children' })).toHaveAttribute('target', '_blank');
  });

  it('Should match with snapshot', () => {
    renderTheme(<MenuLink link="https://google.com">children</MenuLink>);
    expect(screen.getByRole('link', { name: 'children' })).toMatchSnapshot();
  });
});
