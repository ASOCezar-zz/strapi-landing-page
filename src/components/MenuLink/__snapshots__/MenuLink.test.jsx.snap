// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> Should match with snapshot 1`] = `
.c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0A1128;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0.78rem;
  width: 0;
  height: 0.2rem;
  background: #DC143C;
  left: 50%;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c0:hover::after {
  width: 50%;
  left: 25%;
}

<a
  class="c0"
  href="https://google.com"
  target="_self"
>
  children
</a>
`;
